src/main/kotlin/apply/application/EvaluationTargetCsvService.kt

```diff
@@ -0,0 +1,88 @@
+import apply.utils.CsvRow
+import org.apache.commons.csv.CSVFormat
+import org.apache.commons.csv.CSVParser
+import org.apache.commons.csv.CSVRecord
+import org.springframework.stereotype.Service
+import java.io.ByteArrayInputStream
+import java.io.InputStream
+import javax.transaction.Transactional
+
+@Transactional
+@Service
+class EvaluationTargetCsvService(
+    private val evaluationTargetService: EvaluationTargetService,
+    private val evaluationItemRepository: EvaluationItemRepository,
+    private val csvGenerator: CsvGenerator
+) {
+    fun createTargetCsv(evaluationId: Long): ByteArrayInputStream {
+        val targets = evaluationTargetService.findAllByEvaluationIdAndKeyword(evaluationId)
+        val evaluationItems = evaluationItemRepository.findByEvaluationIdOrderByPosition(evaluationId)
+        val evaluationItemHeaders = evaluationItems.map { it.toHeader() }.toTypedArray()
+
+        val headerTitles = arrayOf(ID, NAME, EMAIL, STATUS, *evaluationItemHeaders)
+        val csvRows = targets.map { target ->
+            CsvRow(
+                target.id.toString(),
+                target.name,
+                target.email,
+                target.evaluationStatus.name,
+                *scores(target.answers, evaluationItems).toTypedArray()
+            )
+        }
+        return csvGenerator.generateBy(headerTitles, csvRows)
+    }
+
+    private fun EvaluationItem.toHeader(): String = "${this.title}(${this.maximumScore})"
+
+    private fun scores(
+        answers: List<EvaluationAnswerResponse>,
+        evaluationItems: List<EvaluationItem>
+    ): List<String> {
+        val answerScores = answers.associate { it.evaluationItemId to it.score.toString() }
+        return evaluationItems.map { answerScores[it.id] ?: "0" }
+    }
+
+    fun updateTarget(inputStream: InputStream, evaluationId: Long) {
+        val evaluationItems = evaluationItemRepository.findByEvaluationIdOrderByPosition(evaluationId)
+        inputStream.bufferedReader().use { reader ->
+            val csvParser = CSVParser(
+                reader,
+                CSVFormat.DEFAULT
+                    .withFirstRecordAsHeader()
+                    .withTrim()
+            )
+            for (csvRecord in csvParser) {
+                var targetId = csvRecord.get(ID)
+                var name = csvRecord.get(NAME)
+                var email = csvRecord.get(EMAIL)
+                var evaluationStatus = EvaluationStatus.valueOf(csvRecord.get(STATUS))
+                val evaluationAnswers = readEvaluationAnswers(evaluationItems, csvRecord)
+                // TODO: 평가 대상자 상태 업데이트 기능 구현
+            }
+        }
+    }
+
+    private fun readEvaluationAnswers(
+        evaluationItems: List<EvaluationItem>,
+        csvRecord: CSVRecord
+    ): List<EvaluationAnswer> = evaluationItems.map { evaluationItem ->
+        val score = csvRecord.get(evaluationItem.toHeader())
+        require(score.isNotBlank()) { "평가 항목의 점수에 빈 값이 들어갈 수 없습니다" }
```

Comment:
```suggestion
    ): List<EvaluationAnswer> = evaluationItems.map { evaluationItem ->
        val score = csvRecord.get(evaluationItem.toHeader())
        require(!score.isNullOrBlank()) { "평가 항목의 점수에 빈 값이 들어갈 수 없습니다" }
```

`CSVRecord::get`에 대한 설명입니다
![image](https://user-images.githubusercontent.com/43440195/133913515-95578221-fe5c-4ebd-bc2f-483ecad4a332.png)


`CsvFormat::getNullString`에 따라서 `null`이 리턴될 수도 있다고 하네요. 
저희는 따로 값을 설정해주지 않기 때문에 `null`이 리턴될 거 같아요 😀

자바는 nullable에 대해서 따로 체크하지 않기 때문에 코틀린에서 자바 라이브러리를 사용할 때는 꼭 nullable 여부를 체크해주는 게 좋습니다. 체크하다보면 느끼시겠지만 `@Nullable`이나 주석으로 nullable에 대해서 설명해주는 자바 라이브러리를 만나면 괜히 잘 만들었다는 느낌이 들더라구요.

Reply:
와우 설명 너무 감사합니다..
단순히 빈값으로 인식되는 것만 보고 처리해줬는데 제가 자세히 확인을 못하고 있었네요 😅
생각해보지 못한 부분이었는데 nullable 체크에 유의 명심하겠습니다.
추가로 플랫폼 타입이라는 키워드도 얻어가네요 진짜 덕분에 많이 배웁니다!!! 
꼼꼼한 피드백 감사합니다 !!